libraries/core-react/src/Button/Button.jsx

```diff
@@ -105,36 +98,63 @@ const ButtonBase = styled.button.attrs(({ type = 'button' }) => ({
   }
 `
 
-// TODO:
-// - Missing top spacing
-// - Fallback color?
-// - Use ThemeProvider
-// - Set attr on button
-
-export const Button = ({
-  variant,
-  children,
-  disabled,
-  className,
-  color,
-  ...other
-}) => {
+const RightIcon = styled.span`
+  margin-left: 12px;
+  line-height: normal;
+`
+
+const LeftIcon = styled.span`
+  margin-right: 12px;
+  line-height: normal;
+`
+
+export const Button = forwardRef(function Button(
+  {
+    variant,
+    children,
+    disabled,
+    className,
+    color,
+    rightIcon,
+    leftIcon,
+    href,
+    ...other
+  },
+  ref,
+) {
   const colorBase = colors[color] || {}
   const base = colorBase[variant] || {}
   const baseDisabled = colors.disabled[variant] || {}
 
+  const iconRight = rightIcon && <RightIcon>{rightIcon}</RightIcon>
+  const iconLeft = leftIcon && <LeftIcon>{leftIcon}</LeftIcon>
+  const iconType = variant === 'ghost_icon' ? variant : 'button_icon'
+  const component = href ? 'a' : 'button'
+
+  const baseProps = {
+    ...other,
+    ref,
+    width: button.icon_size[iconType].width,
+    height: button.icon_size[iconType].height,
+  }
+
   return (
     <ButtonBase
+      as={component}
       base={base}
+      type={href ? null : 'button'}
+      href={href ? href : null}
```

Comment:
Think this has to be `undefined` instead of `null` for React to properly drop adding the attributes?